GoogleDataTransport/GDTCORLibrary/GDTCOREvent.m

```diff
@@ -70,12 +70,27 @@ - (void)setDataObject:(id<GDTCOREventDataObject>)dataObject {
   // dispatch_(barrier_ if concurrent)async here, and implement the getter with a dispatch_sync.
   if (dataObject != _dataObject) {
     _dataObject = dataObject;
-    _dataObjectTransportBytes = [dataObject transportBytes];
   }
 }
 
-- (GDTCORStoredEvent *)storedEventWithDataFuture:(GDTCORDataFuture *)dataFuture {
-  return [[GDTCORStoredEvent alloc] initWithEvent:self dataFuture:dataFuture];
+- (bool)writeToURL:(NSURL *)fileURL error:(NSError **)error {
+  NSData *dataTransportBytes = [_dataObject transportBytes];
+  if (dataTransportBytes) {
+    BOOL writingSuccess = [dataTransportBytes writeToURL:fileURL
+                                                 options:NSDataWritingAtomic
+                                                   error:error];
+    if (!writingSuccess) {
+      GDTCORLogError(GDTCORMCEFileWriteError, @"An event file could not be written: %@", fileURL);
+      _fileURL = nil;
+      return NO;
+    } else {
+      _fileURL = fileURL;
+      _dataObject = nil;
+      return YES;
+    }
+  } else {
+    return NO;
+  }
```

Comment:
You might make this easier to read by writing as such:

```
  if (dataTransportBytes == nil) {
    _fileURL = nil;
    _dataObject = nil;
    return NO;
  }
  BOOL writingSuccess = [dataTransportBytes writeToURL:fileURL
                                               options:NSDataWritingAtomic
                                                 error:error];
  if (!writingSuccess) {
    GDTCORLogError(GDTCORMCEFileWriteError, @"An event file could not be written: %@", fileURL);
    _fileURL = nil;
    return NO;
  }
  _fileURL = fileURL;
  _dataObject = nil;
  return YES;
```

Generally, if there's an error condition and early exit you want to try and make that logic apparent. In Swift they have 'guard' statements to help deal with this, but in ObjC we can only write code like this.